### DISCUSSION TITLE
How to visualize feature maps of Encoder?

### DISCUSSION THREAD

**Discussion by vinhlaq** [lukas-blecher/LaTeX-OCR#250] [Q&A]
Hi Lukas, 
I appreciate your work on this project. But I got a question of why ViT+ResNetV2 is better than Pure ViT. And how the input images maintain the spatial information after processed by ResNetV2.
My approach is to access the encoder and visualize the feature map of the last layer of ResNetV2. And also visualize the feature map of the last layer of ViT to have a better insight. 
Could you provide me some codes to plot the feature maps for an image as input?

**Comment by lukas-blecher** [comment]
I'm sure a pure ViT is perfectly useable but I found the hybrid version to be more stable during training. 
Another factor is that for smaller model sizes the hybrid versions actually perform better (see Appendix C in the ViT paper)
The images are actually downscaled by the ResNet, but always by the same factor so it’s no problem for the ViT.

Visualising the resnet embeddings seems reasonable but I don’t have code for that I could share. For the ViT, it’s the attention mask you want to see. The model will most likely only pay attention to the current character in the sequence. I don’t think it’s very interesting to do either but I won’t stop you :)
